mathics/core/builtin.py

```diff
@@ -1133,9 +1133,11 @@ def __init__(self, *args, **kwargs):
 
 class BinaryOperator(Operator):
     """
-    Class for Builtin Binary Operators, e.g. Plus (+)
+    Class for Builtin Infix Operators (which includes Binary
+    operators), e.g. Plus (+).
     """
 
+    # Note: grouping must be Python string, not a Symbol.
```

Comment:
It would be OK if in the class definition, we use a Python string, if we convert it later into a Symbol. So, it would be a problem of `contribute`

Reply:
I checked it out, and indeed `grouping` is used to build a Python `str`, that is parsed as a pattern object for defining formatting and boxing rules. So probably it is OK to keep is as a str.